code/eyetracking/eyetrackingrun.py

```diff
@@ -0,0 +1,781 @@
+                "rxvel_right": "eye1_raw_x_velocity",
+                "ryvel_right": "eye1_raw_y_velocity",
+                "fgxvel": "eye1_fast_gaze_x_velocity",
+                "fgyvel": "eye1_fast_gaze_y_velocity",
+                "fhxyvel": "eye1_fast_head_x_velocity",
+                "fhyvel": "eye1_fast_head_y_velocity",
+                "frxyvel": "eye1_fast_raw_x_velocity",
+                "fryvel": "eye1_fast_raw_y_velocity",
+            }
+        )
+
+        self.samples = self.samples.replace({np.nan: "n/a"})
+        self.samples = self.samples.replace({100000000: "n/a"})
+
+        if self.task_name in ["rest", "bht", "qct"]:
+            output_file_name = f"sub-{self.participant:03d}_ses-{self.session:03d}_task-{self.task_name}_dir-{self.pe}_eyetrack.tsv.gz"
+            output_file_dir = os.path.join(
+                BIDS_folder_path,
+                f"sub-{self.participant:03d}/ses-{self.session:03d}/func/",
+            )
+        elif self.task_name == "dwi":
+            output_file_name = f"sub-{self.participant:03d}_ses-{self.session:03d}_acq-highres_dir-{self.pe}_eyetrack.tsv.gz"
+            output_file_dir = os.path.join(
+                BIDS_folder_path,
+                f"sub-{self.participant:03d}/ses-{self.session:03d}/dwi/",
+            )
+        else:
+            output_file_name = f"sub-{self.participant:03d}_ses-{self.session:03d}_task-{self.task_name}_eyetrack.tsv.gz"
+            output_file_dir = os.path.join(
+                BIDS_folder_path,
+                f"sub-{self.participant:03d}/ses-{self.session:03d}/func/",
+            )
+
+        os.makedirs(output_file_dir, exist_ok=True)
+
+        output_file_full_path = os.path.join(output_file_dir, output_file_name)
+
+        column_order = [
+            "eye_timestamp",
+            "eye1_x_coordinate",
+            "eye1_y_coordinate",
+        ] + [
+            col
+            for col in self.samples.columns
+            if col not in ["eye_timestamp", "eye1_x_coordinate", "eye1_y_coordinate"]
+        ]
+
+        self.samples = self.samples[column_order]
+        self.samples.to_csv(
+            output_file_full_path,
+            sep="\t",
+            index=False,
+            header=False,
+            compression="gzip",
+        )
+
+        return self.samples.columns.tolist()
+
+    def create_info_json(
+        self, BIDS_folder_path: str, info_json_path: str
+    ) -> Optional[str]:
+        """
+        Create and save the info JSON file for eye tracking data.
+
+        Args:
+            BIDS_folder_path (str): Root directory path of the BIDS dataset.
+            info_json_path (str): Path to the info JSON file.
+
+        Returns:
+            Optional[str]: Path to the saved JSON file, or None if no data is available.
+        """
+        with open(info_json_path, "r") as f:
+            info_ET = json.load(f)
+            timestamp_first_trigger = self.find_timestamp_message()
```

Comment:
Please use pathlib, and why the second line needed to be in the context?

```suggestion
        info_ET = json.loads(Path(info_json_path).read_text())
        timestamp_first_trigger = self.find_timestamp_message()
```